### PR TITLE
restart UserEventStream if disconnected after 10 minutes

### PR DESCRIPTION
**Pull request by skynet67**

### CODE REVIEW

File: fitpay/src/main/java/com/fitpay/android/api/services/GenericClient.java [hunk 1]
```diff
@@ -29,6 +31,14 @@ public GenericClient(String baseUrl) {
         client = constructClient(baseUrl, clientBuilder.build());
     }
 
+    public Request.Builder getRequestBuilder(Interceptor.Chain chain){
```
[the remark this answers is not in the document]
Comment (skynet67) [reply fitpay/src/main/java/com/fitpay/android/api/services/GenericClient.java]: No. I'll change it to protected

File: fitpay/src/main/java/com/fitpay/android/api/services/GenericClient.java [hunk 2]
```diff
@@ -29,6 +31,14 @@ public GenericClient(String baseUrl) {
...
+        return chain.request().newBuilder()
+                .header("Accept", "application/json")
+                .header("Content-Type", "application/json")
+                .header("Cache-Control", "max-age=300, no-transform, no-cache")
```
[the remark this answers is not in the document]
Comment (skynet67) [reply fitpay/src/main/java/com/fitpay/android/api/services/GenericClient.java]: agree. max-age should be removed

File: fitpay/src/main/java/com/fitpay/android/api/sse/UserEventStreamManager.java [hunk 3]
```diff
@@ -35,13 +27,17 @@ public static boolean isSubscribed(String userId) {
...
+     *
      * Subscribe to a user event stream, posting SYNC events to the included connector and device when
      * received from the FitPay platform.  The subscription occurs in the background, therefore this
      * method returns a Future for that subscription task.
      *
      * @param userId users Id
      * @return null is possible if not supported
      */
+    @Deprecated
     public static Future<UserEventStream> subscribe(final String userId) {
```
[the remark this answers is not in the document]
Comment (skynet67) [reply fitpay/src/main/java/com/fitpay/android/api/sse/UserEventStreamManager.java]: As we discussed. Java has no overloading

**Comment by climbonbelay** [comment]
User who originally reported this issue pulled in this branch and has some feedback. This user is subscribing to the event stream before opening the webview. The sdk should allow for that. 

> Tried it out and SSE works most of the time. Maybe a conflict because I manually start stream before opening webview, but first time I will still sometimes see a failure

**Pull request merged**